homeassistant/components/august/__init__.py

```diff
@@ -184,23 +190,28 @@ def close_http_session(event):
 
         _LOGGER.debug("August HTTP session closed.")
 
-    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, close_http_session)
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, close_http_session)
     _LOGGER.debug("Registered for Home Assistant stop event")
 
-    return setup_august(hass, config, api, authenticator)
+    token_refresh_lock = asyncio.Lock()
+
+    return await hass.async_add_executor_job(
+        partial(setup_august, hass, config, api, authenticator, token_refresh_lock)
```

Comment:
You only need partial if you want to pass keyword args.
```suggestion
        setup_august, hass, config, api, authenticator, token_refresh_lock
```

Reply:
Will remove